File: week_01_js_recap/problems.js
// Reverse a string
function reverseString(str) {
  return str.split('').reverse().join('');
}

// Find max in array
function findMax(arr) {
  let max = null;
  arr.forEach((n) => (max = n > max ? n : max));
  return max;
}

// Check if palindrome
function isPalindrome(word) {
  const reversed = word.split('').reverse().join('');
  return word === reversed;
}

module.exports = { reverseString, findMax, isPalindrome };

File: week_02_python_intro/problems.py
def reverse_string(s):
    return s[::-1]

def find_max(lst):
    return lst[-1]

def is_palindrome(word):
    pointer1 = 0
    pointer2 = len(word) - 1
    while pointer1 < pointer2:
        if word[pointer1] != word[pointer2]:
            return False
        pointer1 += 1
        pointer2 -= 1
    return True